crates/ra_editor/src/scope/fn_scope.rs

```diff
@@ -241,6 +241,18 @@ struct ScopeData {
     entries: Vec<ScopeEntry>
 }
 
+pub fn resolve_local_name<'a>(name_ref: ast::NameRef, scopes: &'a FnScopes) -> Option<ast::Name<'a>> {
```

Comment:
I think you can return a `ScopeEntry` here, and convert it to a `Name` on the call site.

Reply:
If I return `Option<&'a ScopeEntry>` I can get at the name as a `SmolStr`. I can't get the `Name` without making `ScopeEntry::ast()` public

Reply:
I think it's OK making that public!

Reply:
Done.